avro/src/ser_schema.rs

```diff
@@ -982,31 +986,30 @@ impl<'s, W: Write> SchemaAwareWriteSerializer<'s, W> {
                     )))
                 }
             }
-            Schema::Decimal(decimal_schema) => match decimal_schema.inner.as_ref() {
-                Schema::Bytes => self.write_bytes(value),
-                Schema::Fixed(fixed_schema) => match fixed_schema.size.checked_sub(value.len()) {
-                    Some(pad) => {
-                        let pad_val = match value.len() {
-                            0 => 0,
-                            _ => value[0],
-                        };
-                        let padding = vec![pad_val; pad];
-                        self.writer
-                            .write(padding.as_slice())
-                            .map_err(Details::WriteBytes)?;
-                        self.writer
-                            .write(value)
-                            .map_err(|e| Details::WriteBytes(e).into())
-                    }
-                    None => Err(Details::CompareFixedSizes {
-                        size: fixed_schema.size,
-                        n: value.len(),
+            Schema::Decimal(decimal_schema) => match &decimal_schema.inner {
+                InnerDecimalSchema::Bytes => self.write_bytes(value),
+                InnerDecimalSchema::Fixed(fixed_schema) => {
+                    match fixed_schema.size.checked_sub(value.len()) {
+                        Some(pad) => {
+                            let pad_val = match value.len() {
+                                0 => 0,
+                                _ => value[0],
+                            };
+                            let padding = vec![pad_val; pad];
+                            self.writer
+                                .write(padding.as_slice())
+                                .map_err(Details::WriteBytes)?;
+                            self.writer
+                                .write(value)
+                                .map_err(|e| Details::WriteBytes(e).into())
```

Comment:
The returns only the number of bytes written in the second `.write()` call. IMO it should add the number of the padding length too.